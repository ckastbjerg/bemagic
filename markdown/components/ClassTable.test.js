/* eslint-env jest */
const ClassTable = require('./ClassTable');

test('it throws when missing `elements` parameter', () => {
    expect(() => ClassTable()).toThrow(/elements/);
});

test('it throws when `elements` are missing name property', () => {
    expect(() => ClassTable({ test: {} })).toThrow(/name/);
});

test('it throws when `elements` are missing atRules property', () => {
    expect(() => ClassTable({ test: { name: 'some-name' } })).toThrow(/atRules/);
});

test('it works for correct shape', () => {
    const actual = ClassTable({
        test: {
            name: 'some-name',
            atRules: {
                description: 'Some description...',
            },
        },
    });

    expect(actual).toMatchSnapshot();
});
