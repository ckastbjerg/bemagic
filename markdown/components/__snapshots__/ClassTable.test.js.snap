// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it works for correct shape 1`] = `
"
        | Name | Description |
        |:-----|:-----|
        | **\`some-name\`** | Some description... |

    "
`;
